packages/financial-templates-lib/src/price-feed/HmLandRegistryPriceFeed.ts

```diff
@@ -0,0 +1,253 @@
+import { BN } from "../types";
+import { parseFixed } from "@uma/common";
+import moment from "moment";
+import type { Logger } from "winston";
+import { NetworkerInterface } from "./Networker";
+import { PriceFeedInterface } from "./PriceFeedInterface";
+import Web3 from "web3";
+
+export class HmLandRegistryPriceFeed extends PriceFeedInterface {
+  private readonly uuid: string;
+  private readonly convertPriceFeedDecimals: (number: number | string | BN) => BN;
+  private priceHistory: { date: number; value: BN }[];
+  private currentPrice: BN | null = null;
+  private lastUpdateTime: number | null = null;
+
+  /**
+   * @notice Constructs the TwelveDataApiPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {String} index String used in query to fetch index data, i.e. "URTH"
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Number} priceFeedDecimals Number of priceFeedDecimals to use to convert price to wei.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(
+    private readonly logger: Logger,
+    private readonly index: string,
+    private readonly lookback: number, // lookback should ideally be 90 days
```

Comment:
let's update this comment to 120 days, since ideal lookback needs to be atleast 120 days for UKHPI. Using a lookback of 90 days, gives no results.